.github/workflows/docker-build.yml

```diff
@@ -0,0 +1,76 @@
+name: Source Declarative Manifest - Docker Build
+
+on:
+  push:
+    branches:
+      - main
+      - christo/sdm-test # (dev/test branch)
+    paths:
+      - 'airbyte_cdk/**'
+      - '.github/workflows/docker-build.yml'
+      - 'Dockerfile'
+  workflow_dispatch:
+    inputs:
+      version-tag:
+        description: "Version tag for the image (optional)"
+        required: false
+        type: string
+
+jobs:
+  docker_build:
+    name: Build and Publish SDM Docker Image
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write  # Required for trusted publishing
+      contents: write  # Required for artifact uploads
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up QEMU for mulit-platform builds
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Login to Docker Hub
+        if: ${{ success() && (github.ref == 'refs/heads/main' || github.ref == 'refs/heads/christo/sdm-test' || github.event_name == 'workflow_dispatch') }}
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_PASSWORD }}
+
+      - name: Build test image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          platforms: linux/amd64 # Just build for the runner's architecture during test
+          load: true
+          tags: airbyte/source-declarative-manifest:build-test
+
+      - name: Test image
+        run: |
+          docker run airbyte/source-declarative-manifest:build-test spec 
```

Comment:
I'd suggest separating testing steps and actual publish in different worklows. It makes it easier to maintain the triggers. If you want to share logics between them I'd would suggest leveraging a custom `actions`.

Reply:
Makes sense! Moved the testing steps into their own job and set it as a requirement for publish to run